File: my-fresh-copy/ws/codena/u2/ws13ArraysJson/JavaScript/view/StoreSystem.js
const readline = require("readline");

const ProductView = require("./ProductView");
const ProductController = require("../controller/ProductController");

const view = new ProductView();

const controller = new ProductController(view);

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function mainMenu() {

    view.showMenu();

    rl.question("\nChoose an option: ", (option) => {

        switch (option) {

            case "1":

                rl.question("Enter product name: ", (name) => {

                    rl.question("Enter weight in pounds: ", (weight) => {

                        controller.addProduct(
                            name,
                            parseFloat(weight)
                        );

                        controller.saveProductsToJSON();

                        mainMenu();
                    });
                });

                break;

            case "2":

                controller.readProductsAndConvert();

                mainMenu();

                break;

            case "3":

                console.log("\nProgram finished.");

                rl.close();

                break;

            default:

                console.log("\nInvalid option.");

                mainMenu();
        }
    });
}

mainMenu();

File: my-fresh-copy/ws/codena/u2/ws13ArraysJson/Python/view/store_system.py
from view.product_view import ProductView
from controller.product_controller import ProductController


view = ProductView()

controller = ProductController(view)


def main_menu():

    while True:

        view.show_menu()

        option = input("\nChoose an option: ")

        match option:

            case "1":

                name = input(
                    "Enter product name: "
                )

                weight = float(
                    input(
                        "Enter weight in pounds: "
                    )
                )

                controller.add_product(
                    name,
                    weight
                )

                controller.save_products_to_json()

            case "2":

                controller.read_products_and_convert()

            case "3":

                print("\nProgram finished.")

                break

            case _:

                print("\nInvalid option.")


main_menu()